src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -338,6 +338,43 @@ std::vector<uint8_t> run_and_save_to_memory(std::istringstream & test_idl, std::
     return reader.save_to_memory();
 }
 
+const std::string ConvertTypeToString(ElementType const& type)
+{
+    switch (type)
+    {
+    case ElementType::Void:
+        return "void";
+    case ElementType::Boolean:
+        return "Boolean";
+    case ElementType::Char:
+        return "Char16";
+    case ElementType::I1:
+        return "Int8";
+    case ElementType::U1:
+        return "UInt8";
+    case ElementType::I2:
+        return "Int16";
+    case ElementType::U2:
+        return "UInt16";
+    case ElementType::I4:
+        return "Int32";
+    case ElementType::U4:
+        return "UInt32";
+    case ElementType::I8:
+        return "Int64";
+    case ElementType::U8:
+        return "UInt64";
+    case ElementType::R4:
+        return "Single";
+    case ElementType::R8:
+        return "Double";
+    case ElementType::String:
+        return "String";
+    default:
+        return "";
+    }
+}
+
 void VerifyTypeRefName(std::string const& expectedName, std::string_view currentNamespace, TypeRef const& typeRef)
```

Comment:
Passing string inputs by `std::string_view` provides more flexibility than `std::string`.

Reply:
Also, consider having this function take a `coded_index<TypeDefOrRef>` for flexibility.